src/main/kotlin/com/hypto/iam/server/service/OrganizationsService.kt

```diff
@@ -266,12 +272,17 @@ class OrganizationsServiceImpl : KoinComponent, OrganizationsService {
     }
 
     override suspend fun deleteOrganization(id: String): BaseSuccessResponse {
-        val org = organizationRepo.findById(id) ?: throw EntityNotFoundException("Organization id - $id not found")
-        organizationRepo.deleteById(id)
+        organizationRepo.findById(id) ?: throw EntityNotFoundException("Organization id - $id not found")
 
-        if (org.metadata != null) {
-            val identityGroup = gson.fromJson(org.metadata.data(), IdentityGroup::class.java)
-            identityProvider.deleteIdentityGroup(identityGroup)
+        // TODO: Delete all resources associated with the organization on Cognito
+        txMan.wrap {
+            credentialsRepo.deleteByUserHrn(id)
+            principalPolicyRepo.deleteByPrincipalHrn(id)
+            policyRepo.deleteByOrganizationId(id)
+            userAuthRepo.deleteByUserHrn(id)
+            usersRepo.deleteByOrganizationId(id)
+            passcodeRepo.deleteByOrganizationId(id)
+            organizationRepo.deleteById(id)
```

Comment:
How about the deletion of all the other entities maintained across other services (core, invoice, marketplace etc).

We need a standard way or pass a message for downstream to listen for org deletion.